Update manage-participants playbook

MAHDTech opened the pull request:
This PR adds the following;

- Add null entrypoint for optional usage in the event you want to stop a task from making changes 
- Check callback url & token are both defined, not null and set a default 
- Iterate over correct list of users to remove on ResourceClaim deletion
- Use old requirements format for dependencies to work with older ansible-cli version.
- Align `git_config` variable name with infra-ansible role and set generic default

MAHDTech committed:
Update manage-participants playbook

- Add optional null entrypoint
- Check url & token both defined and not null
- Iterate over correct list of users to remove on ResourceClaim deletion
- Use old galaxy requirements format for dependencies
- Remove the git_config override injected through resource dispatcher in favor of vars at role level
- Remove the use of __meta__ which has been deprecated upstream.
- Update commit msg to include filenames

oybed commented on manage-participants/completion_callback.yml:
```diff
@@ -6,6 +6,14 @@
+    __meta__:
+      callback: {}
+    agnosticd_callback_url: >-
+      {{ __meta__.callback.url | default('') if __meta__ is defined and __meta__.callback is defined else '' }}
```
the `default('')` should be all you need here - i.e.: no need for the if/else logic

oybed replied:
same below

MAHDTech replied:
Done

oybed commented on manage-participants/destroy.yml:
```diff
@@ -15,19 +15,33 @@
+
+    - name: Add all users to removal queue
+      set_fact:
+        lodestar_identities_destroy: "{{ ( lodestar_identities_destroy | default([])) + [ { 'first_name': (identity.first_name | trim), 'last_name': (identity.last_name | trim), 'email': (identity.email | trim) , 'user_name': (identity.email.split('@')[0] | trim), 'state': 'absent' } ] }}"
```
Please work on the readability a bit here - i.e.: newlines, etc. - alternatively look to see if you can perhaps define the user to remove as a yaml object first, e.g.:
```
set_fact:
  user_to_remove:
    first_name: "{{ identify.first_name | trim }}"
    last_name: ...
   : 
```

MAHDTech replied:
Done

oybed commented on manage-participants/process_queue.yml:
```diff
@@ -41,6 +41,12 @@
+          name: Git Bot
+          username: git-bot
+          email: git-bot@no-reply
+          message: "Remove processed jobs from queue"
```
would be great if this could include some more specific info about which one was processed

MAHDTech replied:
@oybed, what did you have in mind, I could add a user removed count?

Adding individual user names here to the commit msg would be messy as it's a single commit for all jobs processed so might end up very long if multiple users are being reset in one execution.

oybed replied:
Was hoping that the job or file name from the queue was easily available to append to this message. Don't think we want to "compose" this, but if already available in a fact, please append it.

MAHDTech replied:
I went with this [example commit msg](https://gitlab.consulting.redhat.com/rht-labs/scratch/lodestar-sandbox/maduncan/engagements/saltlabs/user-reset/iac/-/commit/2eb357a9d686a97cea418d500d2f3ed0ee0c79a0)

oybed commented on manage-participants/completion_callback.yml:
```diff
@@ -48,6 +51,6 @@
         headers:
           Authorization: Bearer {{ agnosticd_callback_token }}
-        validate_certs: "{{ validate_tower_certs | default(true) }}"
+        validate_certs: "{{ validate_tower_certs | default('yes') }}"
```
The quotes should be left off this one since `yes` in this case represents a boolean. Note that Ansible accepts multiple values for booleans to mean the same, e.g.: true, yes, TRUE are all the same, but the string "yes" is generally not (even though it may test to `true` in some cases).

MAHDTech replied:
done

oybed commented on roles/requirements.yml:
```diff
@@ -1,22 +1,6 @@
-    # TODO: Phase 3: Run from release
-    #src: https://github.com/redhat-cop/infra-ansible
-    #version: v2.0.4
+- name: infra-ansible
```
Let's look for a better way to maintain this dependency - e.g.: let's look for a way to use the infra-ansible project that is already set up in tower rather than having a dependency in here that may be though to remember/find when it needs to be updated.

MAHDTech replied:
@oybed, makes sense, do you know of a clean method to use those global projects as the requirements?

Based on what I was able to find they all use recommend the use of `roles/requirements.yml` with tower.

- [Ansible best practices presentation](https://ansible.github.io/workshops/decks/ansible_best_practices.pdf)
- [Galaxy support](https://docs.ansible.com/ansible-tower/latest/html/userguide/projects.html#ansible-galaxy-support)
- [Shared roles](https://www.ansible.com/blog/using-ansible-and-ansible-tower-with-shared-roles)
- [Project collections and Roles](https://www.jeffgeerling.com/blog/2020/ansible-best-practices-using-project-local-collections-and-roles)

I'm looking into perhaps exposing the project directory via `AWX_PROOT_SHOW_PATHS = ['/list/of/', '/paths']` to see if thats possible but will require all paths to be updated.

EDIT: So the path is exposed under /var/lib/awx/projects so I will experiment with using that instead and move this back into draft.

MAHDTech replied:
As discussed, leaving this for a seperate PR

MAHDTech committed:
Update commit msg format

Commit:
Remove all participants

- Ensure all participants are removed when resourceclaim deleted
- Update task name to be more descriptive
- Ensure the queue is also cleared if defined

Commit:
Fix indentation on callback when condition

oybed reviewed (approved):
LGTM
